components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec

```diff
@@ -0,0 +1,191 @@
+#----------------------------------------------------------------------------bh-
+# This RPM .spec file is part of the OpenHPC project.
+#
+# It may have been modified from the default version supplied by the underlying
+# release package (if available) in order to apply patches, perform customized
+# build/install configurations, and supply additional files to support
+# desired integration conventions.
+#
+#----------------------------------------------------------------------------eh-
+
+%include %{_sourcedir}/OHPC_macros
+
+%global gnuver 6.3.0
```

Comment:
The trouble with our targeted Linux distros is that they operate on a bit ancient versions of gcc/g++. Although it may be possible to build LLVM compilers (with all their runtime libraries) using gcc 4.8 (as in Centos 7.3), it does not necessary mean that this build is correct. For example, a part of LLVM project is libomp - OpenMP runtime library which is heavily using compiler builtins and c++11 atomics. This runtime library is built along with (possibly in parallel) the rest of the LLVM project using exactly the same (system?) compiler that builds the other parts. We need to be careful here, as for AArch64 architecture we may face effects of gcc bug like this: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65697
The version selected was 6.3.0 because that was a valid selection before 15th of May 2017 when things changed to 7.1.0 due to commit 6ea6464b0c3115af379734cf08273e56dd5760bb.